src/sasctl/pzmm/write_json_files.py

```diff
@@ -950,3 +982,629 @@ def convertDataRole(self, dataRole):
+
+        package_list = list(pickle_packages) + code_dependencies
+        package_list = list(set(list(flatten(package_list))))
+        package_list = cls.remove_standard_library_packages(package_list)
+        package_and_version = cls.get_local_package_version(package_list)
+        # Identify packages with missing versions
+        missing_package_versions = [
+            item[0] for item in package_and_version if not item[1]
+        ]
+
+        with open(Path(json_path) / "requirements.json", "w") as file:
+            if missing_package_versions:
+                json_step = json.dumps(
+                    [
+                        {
+                            "Warning": "The existence and/or versions for the following packages could not be "
+                            "determined:",
+                            "Packages": ", ".join(missing_package_versions),
+                        }
+                    ],
+                    indent=4,
+                )
+                file.write(json_step)
+
+            for package, version in package_and_version:
+                if version:
+                    json_step = json.dumps(
+                        [
+                            {
+                                "step": "install " + package,
+                                "command": "pip install " + package + "==" + version,
+                            }
+                        ],
+                        indent=4,
+                    )
+                file.write(json_step)
+
+    @classmethod
+    def get_local_package_version(cls, package_list):
+        """
+        Get package_name versions from the local environment. If the package_name
+        does not contain an attribute of "__version__", "version", or
+        "VERSION", no package_name version will be found.
+
+        Parameters
+        ----------
+        package_list : list
+            List of Python packages.
+
+        Returns
+        -------
+        list
+            Nested list of Python package_name names and found versions.
+        """
+
+        def package_not_found_output(package_name, package_versions):
+            print(
+                f"Warning: Package {package_name} was not found in the local environment. Either {package_name} is not "
+                f"a valid Python package, or the package is not present in this environment. The requirements.json file"
+                f" will include a commented out version of the pip installation command at the bottom of the file. "
+                f"Please review the file and verify that the package exists and input the version needed."
+            )
+            package_versions.append([package_name, None])
+            return package_versions
+
+        package_and_version = []
+        import importlib
+
+        for package in package_list:
+            try:
+                name = importlib.import_module(package)
+                try:
```

Comment:
Not necessary, but might be more succinct to do something like:
`version = getattr(package, '__version__', None) or getattr(package, 'version', None) or getattr(package, 'VERSION', None)`